modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/metastorage/pendingtask/DurableBackgroundTask.java

```diff
@@ -37,4 +37,9 @@
      * @param ctx Grid kernal context.
      */
     public void execute(GridKernalContext ctx);
+
+    /**
+     * Callback for cluster deactivation.
+     */
+    public void onClusterDeactivate();
```

Comment:
It would be better if you override `org.apache.ignite.internal.util.worker.GridWorker#cancel` and rename this method to `onCancel()`